wrappers/internal/service_status_test.go

```diff
@@ -0,0 +1,346 @@
+
+	s.systemctlRestorer = systemd.MockSystemctl(func(cmd ...string) ([]byte, error) {
+		s.sysdLog = append(s.sysdLog, cmd)
+		return []byte("ActiveState=inactive\n"), nil
+	})
+	s.delaysRestorer = systemd.MockStopDelays(2*time.Millisecond, 4*time.Millisecond)
+
+	xdgRuntimeDir := fmt.Sprintf("%s/%d", dirs.XdgRuntimeDirBase, os.Getuid())
+	err := os.MkdirAll(xdgRuntimeDir, 0700)
+	c.Assert(err, IsNil)
+	s.agent, err = agent.New()
+	c.Assert(err, IsNil)
+	s.agent.Start()
+}
+
+func (s *serviceStatusSuite) TearDownTest(c *C) {
+	if s.agent != nil {
+		err := s.agent.Stop()
+		c.Check(err, IsNil)
+	}
+	s.systemctlRestorer()
+	s.delaysRestorer()
+	dirs.SetRootDir("")
+	s.DBusTest.TearDownTest(c)
+}
+
+// addSnapServices adds service units for the applications from the snap which
+// are services. The services do not get enabled or started.
+func (s *serviceStatusSuite) addSnapServices(snapInfo *snap.Info, preseeding bool) error {
+	m := map[*snap.Info]*wrappers.SnapServiceOptions{
+		snapInfo: nil,
+	}
+	ensureOpts := &wrappers.EnsureSnapServicesOptions{
+		Preseeding: preseeding,
+	}
+	return wrappers.EnsureSnapServices(m, ensureOpts, nil, progress.Null)
+}
+
+func (s *serviceStatusSuite) TestQueryServiceStatusMany(c *C) {
+	const snapYaml = `name: test-snap
+version: 1.0
+apps:
+  foo:
+    command: bin/foo
+    daemon: simple
+    daemon-scope: user
+  bar:
+    command: bin/bar
+    daemon: simple
+`
+	info := snaptest.MockSnap(c, snapYaml, &snap.SideInfo{Revision: snap.R(1)})
+	fooSrvFile := "snap.test-snap.foo.service"
+	barSrvFile := "snap.test-snap.bar.service"
+
+	r := systemd.MockSystemctl(func(cmd ...string) ([]byte, error) {
+		s.sysdLog = append(s.sysdLog, cmd)
+		if out := systemdtest.HandleMockAllUnitsActiveOutput(cmd, nil); out != nil {
+			return out, nil
+		}
+		if cmd[0] == "--user" && cmd[1] == "show" {
+			return []byte(`Type=simple
+Id=snap.test-snap.foo.service
+Names=snap.test-snap.foo.service
+ActiveState=inactive
+UnitFileState=enabled
+NeedDaemonReload=no
+`), nil
+		}
+		return []byte(`ActiveState=inactive`), nil
+	})
+	defer r()
+
+	err := s.addSnapServices(info, false)
+	c.Assert(err, IsNil)
+
+	sorted := info.Services()
+	sort.Slice(sorted, func(i, j int) bool {
+		return sorted[i].Name < sorted[j].Name
+	})
+
+	sysd := systemd.New(systemd.SystemMode, progress.Null)
+	svcs, usrSvcs, err := internal.QueryServiceStatusMany(sorted, sysd)
+	c.Assert(err, IsNil)
+	c.Assert(svcs, HasLen, 1)
+	c.Check(svcs[0].Name(), Equals, "bar")
+	c.Check(svcs[0].User(), Equals, false)
+	c.Check(svcs[0].ServiceUnitStatus(), DeepEquals, &systemd.UnitStatus{
+		Daemon:           "simple",
+		Id:               "snap.test-snap.bar.service",
+		Name:             "snap.test-snap.bar.service",
+		Names:            []string{"snap.test-snap.bar.service"},
+		Enabled:          true,
+		Active:           true,
+		Installed:        true,
+		NeedDaemonReload: false,
+	})
+	c.Assert(usrSvcs, HasLen, 1)
+	c.Assert(usrSvcs[1000], HasLen, 1)
```

Comment:
1000 seems to be assumed to be the user id running the test, but for example my user id is 1001. We need more mocking or this code needs to use os.Getuid ?